controllers/components/datasciencepipelines/datasciencepipelines_controller_actions.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datasciencepipelines
+
+import (
+	"context"
+	"fmt"
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       DefaultPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(DefaultPath, nil, map[string]string{"namespace": rr.DSCI.Spec.ApplicationsNamespace}); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[0], err)
+	}
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	dsp, ok := rr.Instance.(*componentsv1.DataSciencePipelines)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Ray)", rr.Instance)
+	}
+
+	if dsp.Spec.DevFlags == nil {
+		return nil
+	}
+
+	// Implement devflags support logic
+	// If dev flags are set, update default manifests path
+	if len(dsp.Spec.DevFlags.Manifests) != 0 {
+		manifestConfig := dsp.Spec.DevFlags.Manifests[0]
+		if err := odhdeploy.DownloadManifests(ctx, componentsv1.DataSciencePipelinesComponentName, manifestConfig); err != nil {
+			return err
+		}
+		if manifestConfig.SourcePath != "" {
+			rr.Manifests[0].Path = odhdeploy.DefaultManifestPath
+			rr.Manifests[0].ContextDir = componentsv1.DataSciencePipelinesComponentName
+			rr.Manifests[0].SourcePath = manifestConfig.SourcePath
+		}
+	}
+
+	return nil
```

Comment:
Do we know if there is anything additional needed here looking at the old component implementation I don't think so but I am not sure